lib/chain.js

```diff
@@ -157,19 +157,38 @@ function call(handler, err, req, res, _next) {
         _next(nextErr, req, res);
     }
 
+    function resolve() {
+        next();
+    }
+
+    function reject(error) {
+        if (!error) {
+            error = new errors.VError(
+                {
+                    name: 'AsyncHandlerRejection',
+                    info: { handler: handler._name }
```

Comment:
Please add `cause`

Reply:
what should add here as a `cause`? This error is created whenever the user rejects an async function without passing it anything:

```js
server.get('/params', async (req, res) => {
  return Promise.reject();
});
```
Do you have suggestions for `cause`?

Reply:
You know... just got a mistake in my code... I wasn't wrapping synthetic errors because I thought Restify would throw anything, but actually, if the value passed to `next()` is a string, it will try to route it, for instance:

```js
const restify = require('restify');
const server = restify.createServer({});

server.use(restify.plugins.queryParser({mapParams:true}));

server.get('/route', (req, res, next) => {
   next('getresult');
});

server.get('/string', (req, res, next) => {
   next('invalid');
});

server.get('/number', (req, res, next) => {
   next(1);
});

server.get('/error', (req, res, next) => {
   next(new Error(''));
});

server.get('/result', (req, res, next) => {
   res.send(req.query);
   next();
});

server.listen(8081, 'localhost', function () {
   console.info('server started');
});

```

The result of `curl http://localhost:8081/error?id=1` is:
```json
{"code":"Internal","message":"Error"}
```
The result of `curl http://localhost:8081/number?id=1` is:
```json
{"code":"Internal","message":"1"}
```
The result of `curl http://localhost:8081/string?id=1` is:
```json
{"code":"Internal","message":"RouteMissingError: Route by name doesn't exist"}
```
The result of `curl http://localhost:8081/route?id=1` is:
```json
{"id":"1"}
```

I will update the code to always wrap rejections and update #1837 .

Reply:
FWIW trying to route on strings is IMHO an awful barnacle of Restify, it's worth considering deprecating it--it's essentially a GOTO with all the pain that comes along.

Reply:
wrapping all non-errors in `AsyncHandlerError` actually turned out pretty nice, because if nothing is specified, the cause is `undefined`.

Reply:
and I just discovered `cause` must be an instance of `Error`, so I'm moving the `cause` to inside the `info`.